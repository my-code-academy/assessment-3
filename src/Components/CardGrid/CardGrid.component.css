.CardGrid {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.CardGridAuthor {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
  position: relative;
  border: 7px solid #FB9252;
  border-radius: 25px;
  width: 75%;
  height: 50%;
  margin-left: 10%;
  /* margin-right: 30%; */
  margin-top: 5%; 
}

.CardGridAuthor h1{
  /* margin-left: 40%; */
  text-align: center;
  margin-top: -10px;
  height: 20px;
  line-height: 20px;
  font-size: 15px;
}

.CardGridAuthor h1 span{
  text-align: center;
  background-color: white;
}
