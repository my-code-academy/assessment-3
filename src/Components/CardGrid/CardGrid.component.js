import React, { Component } from 'react';
import './CardGrid.component.css';
import axios from 'axios';
import Card from '../Card/Card.component';

class CardGridComponent extends Component {

  state = {
    booksData: {},
  }

  componentDidMount() {
    axios.get('http://localhost:8080/books').then((booksDataFromAPI) => {
      this.setState({ booksData: booksDataFromAPI.data });
      console.log(this.state.booksData);
    });
  }

  render() {
    return (
      <div className='CardGrid'>
        {
          Object.entries(this.state.booksData).map(author => {
            return <div className='CardGridAuthor'>
              <h1><span>{author[0]}</span></h1>
              {
                author[1].map((bookCard, index) => {
                  return <Card
                    title={bookCard.title}
                    name={bookCard.Name}
                    liked={bookCard.liked}
                    rating={bookCard.rating}
                    imageSrc={`./Images/cover_image.jpeg`}
                  ></Card>
                })
              }
            </div>
          })
        }
      </div>

    )
  }
}

export default CardGridComponent;

