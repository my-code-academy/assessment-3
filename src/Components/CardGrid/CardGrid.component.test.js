import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CardGrid from './CardGrid.component';

describe('CardGrid', () => {
  it ('should match the snapshot', () => {
    const tree = renderer.create(<CardGrid />);
    expect(tree).toMatchSnapshot();
  });
})