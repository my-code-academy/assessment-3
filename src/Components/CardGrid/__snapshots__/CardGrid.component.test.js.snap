// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGrid should match the snapshot 1`] = `
<div
  className="CardGrid"
/>
`;
